staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -784,6 +789,8 @@ func (c *Cacher) dispatchEvents() {
 			// Never send a bookmark event if we did not see an event here, this is fine
 			// because we don't provide any guarantees on sending bookmarks.
 			if lastProcessedResourceVersion == 0 {
+				// pop expired watchers in case there has been no update
+				c.bookmarkWatchers.popExpiredWatchers()
```

Comment:
bookmarkWatchers struct is NOT thread-safe.
I think we should lock here.

Reply:
This PR is changing that